File: 13-RomanToInteger/13-RomanToInteger.py
# Last updated: 04/10/2025, 15:55:11
class Solution:
    def romanToInt(self, s: str) -> int:
        roman={
            'I':1,
            'V':5,
            'X':10,
            'L':50,
            'C':100,
            'D':500,
            'M':1000,
        }
        n=0
        for i in range (len(s)-1):
            if (roman[s[i]]<roman[s[i+1]]):
                n=n-roman[s[i]]
            else:
                n=n+roman[s[i]]
        return n+roman[s[-1]]

File: 13-RomanToInteger/13-RomanToInteger.cpp
// Last updated: 04/10/2025, 15:55:08
class Solution {
public:
    int romanToInt(string s) {
        map<char,int> roman;
        roman.insert(make_pair('I',1));
        roman.insert(make_pair('V',5));
        roman.insert(make_pair('X',10));
        roman.insert(make_pair('L',50));
        roman.insert(make_pair('C',100));
        roman.insert(make_pair('D',500));
        roman.insert(make_pair('M',1000));

        int len = s.length();
        int n,sum = 0;
        for(int i=0;i<len;){
            if((i == len-1) || (roman[s[i]]>=roman[s[i+1]])){
                n = roman[s[i]];
                i++;
            }
            else{
                n = roman[s[i+1]]-roman[s[i]];
                i = i+2;
            }
            sum = sum + n;
        }
        return sum;
    }
};